### PR TITLE
MSPSDS-718: Record incoming question

### PR DESCRIPTION
**Pull request by tom-bell-softwire**

### CODE REVIEW

File: web/app/controllers/create/creation_flow_controller.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
+class Create::CreationFlowController < ApplicationController
```
Comment (broder) [inline web/app/controllers/create/creation_flow_controller.rb]: Would `Investigation::CreationFlowController` be a better name?

File: web/app/controllers/create/creation_flow_controller.rb [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  include FileConcern
+  include Wicked::Wizard
+
+  before_action :set_reporter
```
Comment (broder) [inline web/app/controllers/create/creation_flow_controller.rb]: Do we need `@reporter` for new?

File: web/app/controllers/create_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+class CreateController < ApplicationController
+  before_action :set_available_options
+
+  # GET /create/new
+  def new; end
+
+  # POST /create
+  def create
+    type = request_params[:type]
```
Comment (broder) [inline web/app/controllers/create_controller.rb]: I wonder if we should just put this logic on `investigations#new`. If there's a valid `type` query parameter, then redirect to the relevant controller. If there's no `type` query parameter, render a GET form (with `#new` as the url). What do you think?

File: web/app/views/application/_reporter_type_selection.html.slim [hunk 4]
```diff
@@ -1,43 +1,23 @@
...
+.govuk-grid-row
+  .govuk-grid-column-two-thirds
+    = form_with model: reporter, local: true, url: wizard_path, method: :put do |form|
+      = render "form_validation_display", errors: reporter.errors, model_name: "reporter"
+      fieldset.govuk-fieldset
+        legend.govuk-fieldset__legend.govuk-fieldset__legend--l
+          =  "Who did the question come from?"
+        .govuk-form-group class=("govuk-form-group--error" if reporter.errors[:reporter_type].any?)
+          .govuk-radios
+            - reporter_types = { \
```
Comment (broder) [inline web/app/views/application/_reporter_type_selection.html.slim]: Would this be better in a helper method or controller instance variable?

File: web/app/controllers/create/creation_flow_controller.rb (hunk 2 above)
Comment (tom-bell-softwire) [reply web/app/controllers/create/creation_flow_controller.rb]: Nope!

File: web/app/controllers/investigations_controller.rb [hunk 5]
```diff
@@ -28,7 +28,13 @@ def show
 
   # GET /investigations/new
   def new
-    @investigation = Investigation.new
+    type = params[:type]
```
Comment (broder) [inline web/app/controllers/investigations_controller.rb]: Should we just inline this?

File: web/app/views/layouts/_navbar.html.slim [hunk 6]
```diff
@@ -7,4 +7,5 @@ nav.app-navigation.govuk-clearfix
     li[class=(params[:controller] == "products" && "app-navigation--current-page")]
       = link_to "Products", products_path, "data-topnav": "products"
   .creation
-    = link_to "Create new", new_create_path, class: "govuk-button govuk-button--start", role: "button", draggable: false
+    = link_to "Create new", new_investigation_path, class: "govuk-button govuk-button--start",
+            role: "button", draggable: false
```
Comment (broder) [inline web/app/views/layouts/_navbar.html.slim]: Do we need `draggable` here?

File: web/app/controllers/investigations_controller.rb (hunk 5 above)
Comment (tom-bell-softwire) [reply web/app/controllers/investigations_controller.rb]: Done.

**Pull request merged**